### PR TITLE
Hackdna/s3 static files

### PR DESCRIPTION
**Pull request by hackdna**
For #838

### CODE REVIEW

File: refinery/ui/source/js/analysis-launch/directives/analysis-launch.spec.js [hunk 1]
```diff
@@ -14,16 +14,17 @@ describe('rpAnalysisLaunch directive unit test', function () {
...
     $templateCache.put(
-      '/static/partials/analysis-launch/partials/analysis-launch.html',
+      $window.getStaticUrl('partials/analysis-launch/partials/analysis-launch.html'),
       '<button id="rp-analysis-launch-modal"></button>'
     );
     compile = _$compile_;
     rootScope = _$rootScope_;
     scope = rootScope.$new();
-    template = '<rp-analysis-launch></rp-analysis-launch>';
+    template = angular.element('<rp-analysis-launch></rp-analysis-launch>');
```
Comment (hackdna) [inline refinery/ui/source/js/analysis-launch/directives/analysis-launch.spec.js]: Remove angular.element

File: refinery/ui/source/js/ng-webworker-config.js [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 angular
   .module('ngWebworker')
-  .config(function (WebworkerProvider) {
+  .config(['WebworkerProvider', function (WebworkerProvider) {
     WebworkerProvider.setHelperPath(
-      '/static/vendor/ng-webworker/src/worker_wrapper.min.js'
+      window.getStaticUrl('vendor/ng-webworker/src/worker_wrapper.min.js')
```
Comment (hackdna) [inline refinery/ui/source/js/ng-webworker-config.js]: add comment about refactoring the global getStaticUrl into a service

File: deployment/refinery-modules/refinery/manifests/python.pp [hunk 3]
```diff
@@ -1,11 +1,10 @@
 class refinery::python {
 
   class { '::python':
-    version    => 'system',
-    pip        => true,
-    dev        => true,
-    virtualenv => true,
-    gunicorn   => false,
+    dev        => 'present',
```
Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/python.pp]: I think `version    => 'system',` needs to stick around here

Comment (scottx611x) [inline deployment/refinery-modules/refinery/manifests/python.pp]: Just kidding :) thats a default! Good find

Comment (hackdna) [inline deployment/refinery-modules/refinery/manifests/python.pp]: Yes but even though other params have defaults too, I still had to assign values explicitly for some reason.

File: refinery/ui/source/js/node-mapping/directives/diff-attribute-list.js [hunk 4]
```diff
@@ -9,10 +9,12 @@ function diffAttributeListDirective () {
...
+    templateUrl: function () {
+      return $window.getStaticUrl('partials/node-mapping/directives/diff-attribute-list.html');
+    }
   };
 }
 
 angular
   .module('refineryNodeMapping')
-  .directive('diffAttributeList', diffAttributeListDirective);
+  .directive('diffAttributeList', '$window', diffAttributeListDirective);
```
Comment (jkmarx) [inline refinery/ui/source/js/node-mapping/directives/diff-attribute-list.js]:
```angular
  .module('refineryNodeMapping')
  .directive('diffAttributeList', ['$window', diffAttributeListDirective]);```

**Pull request merged**